### PR TITLE
[Maint] Add dev examples to test_examples.py using subprocess.run

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# References and relevant issues
As noted in https://github.com/napari/napari/pull/6762
the examples under `examples/dev` are not currently tested.

# Description
I tried to naively add `examples/dev` to the tested paths, but there's a problem:
`runpy.run_path` won't execute code under `if __name__ == "__main__":` so most of the dev examples, including the overlays.py which I know is broken just pass.
So I ended up making a new test for the dev ones specifically, to use `subprocess.run` to ensure the scripts are run like a `python example_script.py` would run, including everything under `if __name__ == "__main__":`
This resulted in an issue: the event loop started by `napari.run()` so I added a check if we're testing or not to skip that line for all of the dev examples.

Finally I skip two examples that are rather informative (making graph and things) and not really using anything in napari. They could probably be tested too though...

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
Locally, other than overlays.py -- which I have a fix in the other PR -- I also get a fail with:
`examples/dev/direct-colormap-aliasing-test.py`: 
```
E           napari._tests.test_examples.ExampleError: Dev example failed with:
E           Traceback (most recent call last):
E             File "/Users/piotrsobolewski/Dev/napari/examples/dev/direct-colormap-aliasing-test.py", line 54, in <module>
E               from napari._vispy.layers.labels import build_textures_from_dict, hash2d_get  # noqa
E               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           ImportError: cannot import name 'hash2d_get' from 'napari._vispy.layers.labels' (/Users/piotrsobolewski/Dev/napari/napari/_vispy/layers/labels.py)

```
The `hash2d_get` was removed in https://github.com/napari/napari/pull/6439
cc: @Czaki

File: napari/_tests/test_examples.py [hunk 1]
```diff
@@ -70,6 +77,8 @@ def raise_errors(etype, value, tb):
     monkeypatch.setattr(notification_manager, 'receive_error', raise_errors)
 
     # run the example!
+    # Note: runpy.run_path() does not test anything under
+    # if __name__ == "__main__":
```
Comment (brisvag) [inline napari/_tests/test_examples.py:81]: I think we should rather edit the dev examples to match the non-dev ones (only `napari.run()` under `if name == "__main__"`).

**Review by brisvag — CHANGES_REQUESTED** [review]

**Comment by brisvag** [comment]
I opened a PR (https://github.com/psobolewskiPhD/napari/pull/43) that works well on my machine and does not require a custom trick. I also refactored a bit the testing file so it's a bit more compact and modular.

**Comment by psobolewskiPhD** [comment]
Do you think it's cleaner to just close this an make that PR to napari/napari?

**Pull request closed without merging**